src/librustc_resolve/lib.rs

```diff
@@ -3212,6 +3212,18 @@ impl<'a> Resolver<'a> {
         let name = path[path.len() - 1].node.name;
         // Make sure error reporting is deterministic.
         names.sort_by_key(|name| name.as_str());
+
+
+        // Ugly code, just to see if using case insensitive comparison will help
+        let exact_match = names.iter().find(|x| x.as_str().to_uppercase() == name.as_str().to_uppercase());
+        // do not use Levenstein, just return the value we found (if any)
+        if exact_match.is_some() {
+            return match exact_match {
+                Some(found) => Some(found.clone()),
+                _ => None,
+            }
+        }
+
```

Comment:
It occurs to me that we should likely be making this change as a "tiebreaker" in `syntax::util::lev_distance::find_best_match_for_name` itself: the only time case-insensitive match is going to do _better_ than minimizing Levenshtein distance is when there's a tie for the minimum and the method makes an unlucky choice among the minima, as in the motivating example (`TyInt` and `TyUint` are both Levenshtein-distance 1 away from `TyUInt`).